retrieving/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func main() {
+	client, err := mongo.NewClient(options.Client().ApplyURI("<ATLAS_URI_HERE>"))
+	if err != nil {
+		log.Fatal(err)
+	}
+	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
+	err = client.Connect(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Disconnect(ctx)
+	podcastsCollection := client.Database("quickstart").Collection("podcasts")
+	episodesCollection := client.Database("quickstart").Collection("episodes")
+	var podcast bson.M
+	err = podcastsCollection.FindOne(ctx, bson.M{}).Decode(&podcast)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println(podcast)
+	cursor, err := episodesCollection.Find(ctx, bson.M{"duration": 25})
+	if err != nil {
+		log.Fatal(err)
+	}
+	var episodes []bson.M
+	cursor.All(ctx, &episodes)
+	fmt.Println(episodes)
+	opts := options.Find()
+	opts.SetSort(bson.D{{"duration", -1}})
+	cursor, err = episodesCollection.Find(ctx, bson.D{{"duration", bson.D{{"$gt", 24}}}}, opts)
```

Comment:
I think reusing the `cursor` variable here could be confusing for readers. Maybe split this long function into 3 functions (something like findWithFilter, findOne, findWithOptions)? That would also solve the whitespace issues.

Reply:
I feel like if I introduce functions, it may begin to differ too much from the tutorial. The `main.go` file was more for my own sanity sake. In theory, the tutorial should be everything everyone needs and it should be as bare bones as possible since it is a quick start. Would probably make a lot more sense in a more in depth tutorial.

Reply:
If functions aren't an option, I'd prefer to name the cursors differently. Something like `filterCursor` and `sortCursor`. Especially if we're targeting users that don't have an extensive Go background, re-assigning the `cursor` variable and having the defer statements could confuse them. Naming them differently makes it easier to understand that each cursor is closed separately.